WordPress/gutenberg
# Font Face: Get name from "fontFamily" setting, not "name".

## What?

Get the font-family name from the "fontFamily" field, instead of  the"name" field.

## Why?
To avoid a back-compatibility (BC) break for themes that do not have the `"name"` setting defined.

WP Core also does not require the `"name"` setting. Using it exclusively is a BC break in Core. 

It's not clearly documented that the `"name"` setting is required. Thus, it's been optional.

## How?

Use the font-family's `fontFamily` field instead of  its `name` field.

As the `fontFamily` field could be a comma-separated list of font-families, it parses the list to extract the first entry, using the same approach in the Fonts API.

### Why `fontFamily` setting?

`WP_Theme_JSON` schema requires the `fontFamily` setting in each of the `typography.fontFamilies`. 

## Testing Instructions

1. Use WordPress 6.3, not Core`trunk`. (Why? `trunk` has Font Face in it, so the Font Face files in Gutenberg will not load into memory.) 
2. Activate the Twenty-Twenty-Three (TT3) theme.
3. Open the TT3 `theme.json` file in your favorite editor and modify it as follows:
    * Scroll to line 152 and delete the `"name": "DM Sans",` line.
    * Save.
4. In your browser, open its dev tools and in the HTML inspector, search for and expand the `wp-fonts-local` style element.

Expected:
* Without this PR fix, DM Sans should not be included in the `@font-face` styles ❌ 
* With this PR fix, DM Sans should be included in the `@font-face` styles ✅
